tests/planedata/test_init.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+import pytest
+
+from hexrd.material.crystallography import PlaneData
+
+
+def test_init_with_data_and_from_copy():
+    hkls = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
+    lparms = np.array([1, 1, 1, 1])
+    laueGroup = 'C2h'
+    wavelength = 0.5  # Angstroms
+    strainMag = 0.01
+
+    pd = PlaneData(hkls, lparms, laueGroup, wavelength, strainMag)
+    pd2 = PlaneData(hkls, pd)
+    pd3 = PlaneData(None, pd)
+    pd4 = PlaneData(
+        None,
+        pd,
+        doTThSort=False,
+        exclusions=[False, False, False]
+    )
+    pd5 = PlaneData(
+        None,
+        pd,
+        tThMax=6.0,
+        tThWidth=3.0
+    )
+    assert pd2.hkls.shape == pd3.hkls.shape and np.all(pd2.hkls == pd3.hkls)
+    assert pd2.hkls.shape == pd4.hkls.shape and np.all(pd2.hkls == pd4.hkls)
+
+
+def test_init_with_invalid_params():
+    # One extra parameter but not a PlaneData object
+    with pytest.raises(Exception):
+        PlaneData(np.array([[1, 0, 0], [0, 1, 0]]), 1)
+    # Invalid number of parameters
+    with pytest.raises(Exception):
+        PlaneData(
+            np.array([[1, 0, 0], [0, 1, 0]]), np.array([1, 1, 1, 1, 1]), 3
+        )
+    # Invalid number of unnamed parameters
+    with pytest.raises(Exception):
+        PlaneData(
+            np.array([[1, 0, 0], [0, 1, 0]]),
+            np.array([1, 1, 1, 1]),
+            'C2h',
+            0.5,
+            doTThSort=True,
+        )
+    # Invalid named parameter
+    with pytest.raises(Exception):
```

Comment:
It's usually nice for these to catch specific kinds of exceptions, rather than just `Exception` (which could be caused by anything). But if that's what this raises, then maybe that's okay.

Reply:
Well it's a bit weird currently, invalid arguments could raise a NotImplementedError if they are in *args, but will raise a RuntimeError if they are in **kwargs.  I kind of figured that might change so I didn't want to specify.  But I'll check for it being one of the two